src/main/java/com/gmail/buer2012/controller/AuthController.java

```diff
@@ -148,10 +148,17 @@ private void sendConfirmationEmail(String requestEmail, User user) {
     @PostMapping("/confirmRestore")
     public ResponseEntity<?> restorePassword(@RequestBody RestorePasswordRequest restorePasswordRequest) {
         EmailConfirmationToken emailConfirmationToken = emailConfirmationTokenRepository.findByToken(restorePasswordRequest.getToken());
-        User user = emailConfirmationToken.getUser();
-        emailConfirmationTokenRepository.delete(emailConfirmationToken);
-        user.setPassword(passwordEncoder.encode(restorePasswordRequest.getPassword()));
-        userService.updateUser(user);
-        return ResponseEntity.ok(0);
+        if(emailConfirmationToken != null) {
+            User user = emailConfirmationToken.getUser();
+            emailConfirmationTokenRepository.delete(emailConfirmationToken);
+            user.setPassword(
+                passwordEncoder.encode(restorePasswordRequest.getPassword()));
+            userService.updateUser(user);
+            return ResponseEntity.ok(0);
```

Comment:
Why always using 0? Can't it be left blank?

Reply:
It must have some body

Reply:
`return ResponseEntity.ok().build();`